packages/jest-config/src/normalize.js

```diff
@@ -237,6 +238,19 @@ const normalizeMissingOptions = (options: InitialOptions): InitialOptions => {
       .digest('hex');
   }
 
+  if (Array.isArray(options.projects)) {
+    options.projects = options.projects.map((project, index) => {
+      if (typeof project !== 'string' && !project.hasOwnProperty('name')) {
+        project.name = crypto
+          .createHash('md5')
+          .update(project.rootDir || `${options.rootDir}/${index}`)
```

Comment:
using a slash here as separator isn't a great idea. I recommend using another delimiter that is invalid in paths (like `:`).